module/VuFind/src/VuFind/Recommend/ConsortialVuFind.php

```diff
@@ -0,0 +1,241 @@
+        $limitSetting = intval($settings[1] ?? 0);
+        $this->limit = $limitSetting > 0 ? $limitSetting : $this->limit;
+        $configSectionName = $settings[2] ?? 'ReShare';
+
+        // Read config file
+        $configSection = $this->config->get($configSectionName);
+        if ($configSection) {
+            $this->resultsBaseUrl = $configSection->results_base_url;
+            $this->recordBaseUrl = $configSection->record_base_url;
+            $this->searchFilters = $configSection->filters?->toArray() ?? [];
+
+            // Configure connection
+            $this->connection->setBaseUrl($configSection->api_base_url);
+
+            // Confirm that required configuration is present
+            $this->hasMinimumConfig = $this->resultsBaseUrl
+                && $this->recordBaseUrl
+                && $configSection->api_base_url;
+            if (!$this->hasMinimumConfig) {
+                $this->logError("Required configuration missing in '$configSectionName'
+                    section of ConsortialVuFind.ini.");
+            }
+        } else {
+            $this->logError("'$configSectionName' section not found in ConsortialVuFind.ini.");
+        }
+    }
+
+    /**
+     * Called before the Search Results object performs its main search
+     * (specifically, in response to \VuFind\Search\SearchRunner::EVENT_CONFIGURED).
+     * This method is responsible for setting search parameters needed by the
+     * recommendation module and for reading any existing search parameters that may
+     * be needed.
+     *
+     * @param \VuFind\Search\Base\Params $params  Search parameter object
+     * @param \Laminas\Stdlib\Parameters $request Parameter object representing user
+     * request.
+     *
+     * @return void
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function init($params, $request)
+    {
+        // No action needed.
+    }
+
+    /**
+     * Called after the Search Results object has performed its main search. This
+     * may be used to extract necessary information from the Search Results object
+     * or to perform completely unrelated processing.
+     *
+     * @param \VuFind\Search\Base\Results $results Search results object
+     *
+     * @return void
+     */
+    public function process($results)
+    {
+        $this->queryString = $results->getParams()->getQuery()->getString();
+    }
+
+    /**
+     * Get the consortial VuFind instance's search results.
+     *
+     * @return array
+     */
+    public function getResults()
+    {
+        if (!$this->hasMinimumConfig) {
+            return [];
+        }
+
+        $results = $this->connection->search(
+            $this->queryString,
+            $this->requestParam,
+            $this->limit,
+            $this->searchFilters
+        );
+        foreach (($results['records'] ?? []) as $i => $record) {
+            $results['records'][$i]['url'] = $this->recordBaseUrl . '/' . $record['id'];
```

Comment:
I think we need to URLencode `$record['id']` here to cover things like the edge case where IDs contain slashes.

Reply:
Thanks for the catch, I was thinking of it just as a number but that's obviously wrong.